### PR TITLE
feat: prevent default identity provider to be created

### PR DESCRIPTION
**Pull request by podlesrafal**
Feat: AM-3443

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-distribution/src/main/resources/config/gravitee.yml [hunk 1]
```diff
@@ -474,3 +474,10 @@ liquibase:
 #    algorithm: None
 #    properties:
 #      rounds: 10
+#domains:
+#  reporters:
+#    default:
+#      enabled: true
+#  identities:
+#    default:
+#      enabled: true
```
Comment (leleueri) [inline gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-distribution/src/main/resources/config/gravitee.yml:483]: **question:** is it really useful to have this sections in the GW gravitee.yaml

**Review by leleueri — CHANGES_REQUESTED** [review]
TODO: can you please manage this new parameter in the helm chart ? https://github.com/gravitee-io/gravitee-access-management/blob/master/helm/templates/api/api-configmap.yaml#L276

**Review by leleueri — CHANGES_REQUESTED** [review]

File: gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/DefaultIdentityProviderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+        return mongoUri;
+    }
+
+
+    private String createProviderConfig(String referenceId, NewIdentityProvider identityProvider) {
+        final Map<String, Object> configMap = createProviderConfiguration(referenceId, identityProvider);
+        try {
+            return new ObjectMapper().writeValueAsString(configMap);
+        } catch (JsonProcessingException e) {
```
Comment (lgw-gravitee) [inline gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/DefaultIdentityProviderServiceImpl.java:196]: ObjectMapper could be a dependecy on the DefaultIdentityProviderServiceImpl

Comment (podlesrafal) [reply gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/DefaultIdentityProviderServiceImpl.java:196]: There is no Bean created with ObjectMapper in this package.

File: gravitee-am-management-api/gravitee-am-management-api-standalone/gravitee-am-management-api-standalone-distribution/src/main/resources/config/gravitee.yml [hunk 3]
```diff
@@ -259,6 +259,9 @@ domains:
 #      # should the default (database) reporter be created. E.g. if the organization defines a global reporter,
 #      # domain-level reporters might not be necessary
 #      enabled: true
+#  identities:
+#    default:
+#      enabled: true
```
[the remark this answers is not in the document]
Comment (podlesrafal) [reply gravitee-am-management-api/gravitee-am-management-api-standalone/gravitee-am-management-api-standalone-distribution/src/main/resources/config/gravitee.yml]: done

File: helm/templates/api/api-configmap.yaml [hunk 4]
```diff
@@ -296,6 +296,10 @@ data:
       reporters:
         {{- .Values.api.domains.reporters | toYaml | nindent 8 }}
       {{- end }}
+      {{- if .Values.api.domains.identities }}
+      identities:
+        {{- .Values.api.domains.identities | toYaml | nindent 8 }}
+      {{- end }}
```
[the remark this answers is not in the document]
Comment (podlesrafal) [reply helm/templates/api/api-configmap.yaml]: done

**Review by leleueri — APPROVED** [review]

File: gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/DefaultIdentityProviderServiceImpl.java (hunk 2 above)
Comment (lgw-gravitee) [reply gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/DefaultIdentityProviderServiceImpl.java:197]: ObjectMapper creation can be expensive (it loads modules etc). How about adding it as final mapper = new ObjectMapper at class level?

**Review by lgw-gravitee — APPROVED** [review]

**Pull request merged**